File: src/main/scala/com/martomate/tripaint/view/gui/MainStageButtons.scala
package com.martomate.tripaint.view.gui

import com.martomate.tripaint.view.{MenuBarAction, TriPaintViewListener}
import scalafx.scene.input.{KeyCode, KeyCodeCombination, KeyCombination}

class MainStageButtons(control: TriPaintViewListener) {
  val New: MenuBarAction = MenuBarAction("New", "new", new KeyCodeCombination(KeyCode.N, KeyCombination.ControlDown), control.action_new())
  val Open: MenuBarAction = MenuBarAction("Open", "open", new KeyCodeCombination(KeyCode.O, KeyCombination.ControlDown), control.action_open())
  val OpenHexagon: MenuBarAction = MenuBarAction("Open hexagon", "open_hexagon", new KeyCodeCombination(KeyCode.H, KeyCombination.ControlDown), control.action_openHexagon())
  val Save: MenuBarAction = MenuBarAction("Save", "save", new KeyCodeCombination(KeyCode.S, KeyCombination.ControlDown), control.action_save())
  val SaveAs: MenuBarAction = MenuBarAction("Save As", accelerator = new KeyCodeCombination(KeyCode.S, KeyCombination.ControlDown, KeyCombination.ShiftDown), onAction = control.action_saveAs())
  val Exit: MenuBarAction = MenuBarAction("Exit", onAction = control.action_exit())
  val Undo: MenuBarAction = MenuBarAction("Undo", "undo", new KeyCodeCombination(KeyCode.Z, KeyCombination.ControlDown), control.action_undo())
  val Redo: MenuBarAction = MenuBarAction("Redo", "redo", new KeyCodeCombination(KeyCode.Y, KeyCombination.ControlDown), control.action_redo())
  val Cut: MenuBarAction = MenuBarAction("Cut", "cut")
  val Copy: MenuBarAction = MenuBarAction("Copy", "copy")
  val Paste: MenuBarAction = MenuBarAction("Paste", "paste")
  val Move: MenuBarAction = MenuBarAction("Move", "move")
  val Scale: MenuBarAction = MenuBarAction("Scale", "scale")
  val Rotate: MenuBarAction = MenuBarAction("Rotate", "rotate")
  val Blur: MenuBarAction = MenuBarAction("Blur", onAction = control.action_blur())
  val MotionBlur: MenuBarAction = MenuBarAction("Motion blur", onAction = control.action_motionBlur())
  val RandomNoise: MenuBarAction = MenuBarAction("Random noise", onAction = control.action_randomNoise())
  val Scramble: MenuBarAction = MenuBarAction("Scramble", onAction = control.action_scramble())
}

File: ui/src/main/kotlin/tripaint/view/gui/MainStageButtons.kt
package tripaint.view.gui

import javafx.scene.input.KeyCode
import javafx.scene.input.KeyCodeCombination
import javafx.scene.input.KeyCombination
import tripaint.view.MenuBarAction

object MainStageButtons {
    val New: MenuBarAction = MenuBarAction(
        "New",
        "new",
        KeyCodeCombination(KeyCode.N, KeyCombination.CONTROL_DOWN),
    UIAction.New
    )
    val Open: MenuBarAction = MenuBarAction(
        "Open",
        "open",
        KeyCodeCombination(KeyCode.O, KeyCombination.CONTROL_DOWN),
    UIAction.Open
    )
    val OpenHexagon: MenuBarAction = MenuBarAction(
        "Open hexagon",
        "open_hexagon",
        KeyCodeCombination(KeyCode.H, KeyCombination.CONTROL_DOWN),
    UIAction.OpenHexagon
    )
    val Save: MenuBarAction = MenuBarAction(
        "Save",
        "save",
        KeyCodeCombination(KeyCode.S, KeyCombination.CONTROL_DOWN),
    UIAction.Save
    )
    val SaveAs: MenuBarAction = MenuBarAction(
        "Save As",
        accelerator =
        KeyCodeCombination(KeyCode.S, KeyCombination.CONTROL_DOWN, KeyCombination.SHIFT_DOWN),
    action = UIAction.SaveAs
    )
    val Exit: MenuBarAction = MenuBarAction("Exit", action = UIAction.Exit)
    val Undo: MenuBarAction = MenuBarAction(
        "Undo",
        "undo",
        KeyCodeCombination(KeyCode.Z, KeyCombination.CONTROL_DOWN),
    UIAction.Undo
    )
    val Redo: MenuBarAction = MenuBarAction(
        "Redo",
        "redo",
        KeyCodeCombination(KeyCode.Y, KeyCombination.CONTROL_DOWN),
    UIAction.Redo
    )
    val Cut: MenuBarAction = MenuBarAction("Cut", "cut")
    val Copy: MenuBarAction = MenuBarAction("Copy", "copy")
    val Paste: MenuBarAction = MenuBarAction("Paste", "paste")
    val Move: MenuBarAction = MenuBarAction("Move", "move")
    val Scale: MenuBarAction = MenuBarAction("Scale", "scale")
    val Rotate: MenuBarAction = MenuBarAction("Rotate", "rotate")
    val Blur: MenuBarAction = MenuBarAction("Blur", action = UIAction.Blur)
    val MotionBlur: MenuBarAction =
        MenuBarAction("Motion blur", action = UIAction.MotionBlur)
    val RandomNoise: MenuBarAction =
        MenuBarAction("Random noise", action = UIAction.RandomNoise)
    val Scramble: MenuBarAction = MenuBarAction("Scramble", action = UIAction.Scramble)
    val Cell: MenuBarAction = MenuBarAction("Cell", action = UIAction.Cell)
}